cirq-google/cirq_google/optimizers/two_qubit_gates/gate_compilation.py

```diff
@@ -1,302 +1,46 @@
 # pylint: disable=wrong-or-nonexistent-copyright-notice
 """Attempt to tabulate single qubit gates required to generate a target 2Q gate
-with a product A k A."""
-from functools import reduce
-from typing import Tuple, Sequence, List, NamedTuple
+with a product A k A.
+
+Attention: this module is now deprecated! Use the classes from cirq-core instead."""
+from typing import Tuple
 
 from dataclasses import dataclass
 import numpy as np
 
 import cirq
-from cirq import value
-from cirq._compat import proper_repr, proper_eq
-from cirq_google.optimizers.two_qubit_gates.math_utils import (
-    kak_vector_infidelity,
-    vector_kron,
-    weyl_chamber_mesh,
-    random_qubit_unitary,
-    kak_vector_to_unitary,
+from cirq import (
+    TwoQubitGateTabulation,
+    TwoQubitGateTabulationResult,
+    two_qubit_gate_product_tabulation,
 )
+from cirq._compat import deprecated, deprecated_class
 
 _SingleQubitGatePair = Tuple[np.ndarray, np.ndarray]
 
 
-class TwoQubitGateCompilation(NamedTuple):
-    r"""Represents a compilation of a target 2-qubit with respect to a base
-    gate.
-
-    This object encodes the relationship between 4x4 unitary operators
-
-    U_target ~ k_N · U_base · k_{N-1} · ... · k_1 · U_base · k_0
-
-    where U_target, U_base are 2-local and k_j are 1-local.
-
-    Attributes:
-        base_gate: 4x4 unitary denoting U_base above.
-        target_gate: 4x4 unitary denoting U_target above.
-        local_unitaries: Sequence of 2-tuples
-            $(k_{00}, k_{01}), (k_{10}, k_{11}) \ldots$ where
-            $k_j = k_{j0} \otimes k_{j1}$ in the product above.
-            Each $k_{j0}, k_{j1}$ is a 2x2 unitary.
-        actual_gate: 4x4 unitary denoting the right hand side above, ideally
-            equal to U_target.
-        success: Whether actual_gate is expected to be close to U_target.
-    """
-    base_gate_unitary: np.ndarray
-    target_gate: np.ndarray
-    local_unitaries: Tuple[_SingleQubitGatePair, ...]
-    actual_gate: np.ndarray
-    success: bool
+@deprecated_class(
+    deadline='v0.16',
+    fix='Use cirq.TwoQubitGateTabulationResult instead.',
+    name='TwoQubitGateCompilation',
+)
+class TwoQubitGateCompilation(TwoQubitGateTabulationResult):
+    pass
 
 
 @dataclass
```

Comment:
IIUC, we can remove this `@dataclass` decorator since the class now simply inherits from the parent dataclass.

Reply:
Done